sampler.go

```diff
@@ -9,10 +9,70 @@ import (
 
 // Sampler returns the next search points
 type Sampler interface {
-	// Sample a parameter for a given distribution.
+	// Sample a single parameter for a given distribution.
 	Sample(*Study, FrozenTrial, string, interface{}) (float64, error)
 }
 
+// RelativeSampler returns the next search points
+type RelativeSampler interface {
```

Comment:
The interface is little different with Optuna's one. RelativeSampler does not provide sample_independent method.